lib/openstudio-standards/standards/Standards.ScheduleRuleset.rb

```diff
@@ -873,4 +906,467 @@ def process_hrs_of_operation_hash(sch_day, hoo_start, hoo_end, val_flr, val_clg,
 
   end
 
+  # Create a sequential array of values from a ScheduleReset object
+  # Will actually include 24 extra values if model year is a leap year
+  # Will also include 24 values at end of array representing the holiday day schedule
+  # Intended use is for storing fan schedules to hash of ZoneName:SchedArray so schedules
+  # can be saved when HVAC objects are deleted
+  # @author Doug Maddox, PNNL
+  # @param [object] model 
+  # @param [Object] schedule_ruleset
+  # @return [Array<Double>] Array of sequential hourly values for year + 24 hours at end for holiday
+  def get_8760_values_from_schedule_ruleset(model, schedule_ruleset)
+    yd = model.getYearDescription
+    start_date = yd.makeDate(1, 1)
+    end_date = yd.makeDate(12, 31)
+
+    day_of_week = start_date.dayOfWeek.valueName
+
+    values = OpenStudio::DoubleVector.new
+    day = OpenStudio::Time.new(1.0)
+    interval = OpenStudio::Time.new(1.0 / 24.0)
+    day_schedules = schedule_ruleset.to_ScheduleRuleset.get.getDaySchedules(start_date, end_date)
+
+    numdays = day_schedules.size
+
+    # Get holiday schedule and append to end of values array
+    day_schedule_holiday = nil
+    schedule_ruleset.to_ScheduleRuleset.get.scheduleRules.each do |week_rule|
+      # If a holiday day type is defined, then store the schedule object for the first occurrence
+      # For now this is not implemented into the 8760 array
+      #if week_rule.applyHoliday
+      #  day_schedule_holiday = week_rule.daySchedule
+      #  break
+      #end
```

Comment:
We might want to remove these comments before merging.

Reply:
I keep this as is in case they will be useful in the future.